qiskit/providers/basic_provider/basic_simulator.py

```diff
@@ -500,6 +505,98 @@ def _run_job(self, job_id: str, run_input) -> Result:
 
         return Result.from_dict(result)
 
+    def _is_clifford_circuit(self, circuit: QuantumCircuit):
+        """Check if circuit is Clifford and return Clifford object or None."""
+        try:
+            circ_no_meas = circuit.remove_final_measurements(inplace=False)
+            return Clifford(circ_no_meas)
+        except QiskitError:
+            return None
```

Comment:
This is really a rather expensive check, and it runs on every circuit - `remove_final_measurements` is not cheap, nor is constructing a Clifford object.  We probably need to revisit this.